### PR TITLE
Fixes #1906

### LINKED ISSUE #1906: Bond stereo information not output to SMARTS
**Description:**
Here's the reproducible:
```
In [2]: m = Chem.MolFromSmiles('C/C=C/C')

In [3]: Chem.MolToSmiles(m)
Out[3]: 'C/C=C/C'

In [4]: Chem.MolToSmarts(m)
Out[4]: '[#6]-[#6]=[#6]-[#6]'
```

From this mailing list thread:
https://www.mail-archive.com/rdkit-discuss@lists.sourceforge.net/msg07994.html

### PR DESCRIPTION
**Pull request by greglandrum**
includes a minor bit of refactoring/reduction of duplicated code

### CODE REVIEW

File: Code/GraphMol/SmilesParse/SmartsWrite.cpp [hunk 1]
```diff
@@ -333,6 +333,40 @@ std::string getRecursiveStructureQuerySmarts(
...
+      res = "=";
+      break;
+    case Bond::TRIPLE:
+      res = "#";
+      break;
+    case Bond::AROMATIC:
+      res = ":";
+      break;
+    case Bond::DATIVE:
+      if (!reverseDative)
```
[the remark this answers is not in the document]
Comment (greglandrum) [reply Code/GraphMol/SmilesParse/SmartsWrite.cpp]: "just lazy"

File: Code/GraphMol/SmilesParse/SmartsWrite.cpp [hunk 2]
```diff
@@ -333,6 +333,40 @@ std::string getRecursiveStructureQuerySmarts(
   return res;
 }
 
+std::string getBasicBondRepr(Bond::BondType typ, Bond::BondDir dir,
+                             bool doIsomericSmiles, bool reverseDative) {
+  std::string res;
+  switch (typ) {
+    case Bond::SINGLE:
+      if (dir == Bond::ENDDOWNRIGHT && doIsomericSmiles) {
```
[the remark this answers is not in the document]
Comment (greglandrum) [reply Code/GraphMol/SmilesParse/SmartsWrite.cpp]: yeah, I'll switch that

File: Code/GraphMol/SmilesParse/SmartsWrite.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bp-kelley) [reply Code/GraphMol/SmilesParse/SmartsWrite.cpp]: one liner for the win?
```
res = (reverseDative) ? "<-" : "->"; break;
```

**Comment by bp-kelley** [comment]
Happy to merge, some stylistic comments only.

**Pull request merged**